pkg/lockfile/parse-go-lock.go

```diff
@@ -82,6 +82,20 @@ func (e GoLockExtractor) Extract(f DepFile) ([]PackageDetails, error) {
 		}
 	}
 
+	if parsedLockfile.Go != nil && parsedLockfile.Go.Version != "" {
+		goVersion := parsedLockfile.Go.Version
+		versionParts := strings.Split(goVersion, ".")
```

Comment:
maybe we should use some Go package that deals with the versioning here?
I found this package: https://pkg.go.dev/golang.org/x/mod/semver, but this requires a "v" prefix in the version string.

Reply:
If only we had a robust semantic based version parsing and comparing library already... 😁

(`internal/semantic` should have everything you need for this)

Reply:
Updated, thanks!